cmd/nerdctl/container_logs_test.go

```diff
@@ -206,3 +207,30 @@ func TestLogsWithForegroundContainers(t *testing.T) {
 		}(t)
 	}
 }
+
+func TestTailFollowRotateLogs(t *testing.T) {
+	t.Parallel()
+	if runtime.GOOS == "windows" {
+		t.Skip("tail log is not supported on Windows")
+	}
+	base := testutil.NewBase(t)
+	containerName := testutil.Identifier(t)
+
+	const sampleJSONLog = `{"log":"A\n","stream":"stdout","time":"2024-04-11T12:01:09.800288974Z"}`
+	const linesPerFile = 100
+
+	defer base.Cmd("rm", "-f", containerName).Run()
+	base.Cmd("run", "-d", "--log-driver", "json-file",
+		"--log-opt", fmt.Sprintf("max-size=%d", len(sampleJSONLog)*linesPerFile),
+		"--log-opt", "max-file=10",
+		"--name", containerName, testutil.CommonImage,
+		"sh", "-euc", "while true; do echo A; done").AssertOK()
+
+	tailLogCmd := base.Cmd("logs", "-f", containerName)
+	tailLogCmd.Timeout = 100 * time.Millisecond
```

Comment:
how to ensure that a rotation occurs just before 100ms ?

Reply:
HI @fahedouch 

The Tescase is to test whether `nerdctl logs -f` works correctly on the rotation file. And it does not test whether the `rotation` can work correctly. 

The rotation will work when the file size exceeds `len(sampleJSONLog)*linesPerFile).`
So if the log size is insufficient for rotation, the assert `assert.Equal(t, true, len(tailLogs) > linesPerFile)` cannot pass.
So the rotation must be occurs :-)

-----
The test in the Intel environment:
* 3740 lines are printed in 100ms

And they are rotated to 10 files.
```
root@kay201:/var/lib/nerdctl/1935db59/containers/nerdctl-test/1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff# ls
1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff-json.log
1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff-json.log.640
1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff-json.log.641
1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff-json.log.642
1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff-json.log.643
1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff-json.log.644
1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff-json.log.645
1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff-json.log.646
1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff-json.log.647
1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff-json.log.648
hostname
log-config.json
oci-hook.startContainer.log
resolv.conf
```

How do you think about that :-)

Reply:
> So if the log size is insufficient for rotation, the assert assert.Equal(t, true, len(tailLogs) > linesPerFile) cannot pass.

good but to be more accurate I suggest 

```
assert.Equal(t, true, len(tailLogs) > len(sampleJSONLog)*linesPerFile)
```

Reply:
> > So if the log size is insufficient for rotation, the assert assert.Equal(t, true, len(tailLogs) > linesPerFile) cannot pass.
> 
> good but to be more accurate I suggest
> 
> ```
> assert.Equal(t, true, len(tailLogs) > len(sampleJSONLog)*linesPerFile)
> ```

HI @fahedouch 

Thanks for review :-)

The output of `nerdctl logs -f ` is : 
```
root@kay201:~# nerdctl -n nerdctl-test logs 1383c701a5cf
A
A
A
....
```
So the variable `tailLogs` is : ["A","A","A" .....] . 


The content of JSON file logs is :
```
root@kay201:/var/lib/nerdctl/1935db59/containers/nerdctl-test/1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff# cat 1383c701a5cf46cce2d122ae5808afe8416f8ce43e32d61424830a382c7847ff-json.log
{"log":"A\n","stream":"stdout","time":"2024-04-12T01:49:07.70654089Z"}
{"log":"A\n","stream":"stdout","time":"2024-04-12T01:49:07.706873857Z"}
{"log":"A\n","stream":"stdout","time":"2024-04-12T01:49:07.706881216Z"}
{"log":"A\n","stream":"stdout","time":"2024-04-12T01:49:07.706884883Z"}
{"log":"A\n","stream":"stdout","time":"2024-04-12T01:49:07.706888406Z"}
{"log":"A\n","stream":"stdout","time":"2024-04-12T01:49:07.706891915Z"}
```

So it's why `len(tailLogs) ` and `linesPerFile` are comparable, And I think `len(tailLogs) > len(sampleJSONLog)*linesPerFile` is not needed because they are not comparable:-)

But if we want the comparison to be more strict， maybe `len(tailLogs) > 2*linesPerFile` is OK.

Does it OK?

Reply:
It is fine ;)